From 065e12983c152222e11601ed4eebbcc021d95646 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Thu, 27 Aug 2015 20:23:47 +0800
Subject: [PATCH] MLK-11438-2 epdc: enable lpsr mode

restore the pinmux when resume from LPSR mode in suspend

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 drivers/video/mxc/mxc_epdc_v2_fb.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/mxc/mxc_epdc_v2_fb.c b/drivers/video/mxc/mxc_epdc_v2_fb.c
index 86e59ae..b9e99ba 100644
--- a/drivers/video/mxc/mxc_epdc_v2_fb.c
+++ b/drivers/video/mxc/mxc_epdc_v2_fb.c
@@ -5676,10 +5676,13 @@ static int mxc_epdc_fb_suspend(struct device *dev)
 
 	data->pwrdown_delay = FB_POWERDOWN_DISABLE;
 	ret = mxc_epdc_fb_blank(FB_BLANK_POWERDOWN, &data->info);
+
 	if (ret)
 		goto out;
 
 out:
+	pinctrl_pm_select_sleep_state(dev);
+
 	return ret;
 }
 
@@ -5687,6 +5690,8 @@ static int mxc_epdc_fb_resume(struct device *dev)
 {
 	struct mxc_epdc_fb_data *data = dev_get_drvdata(dev);
 
+	pinctrl_pm_select_default_state(dev);
+
 	mxc_epdc_fb_blank(FB_BLANK_UNBLANK, &data->info);
 	epdc_init_settings(data);
 	data->updates_active = false;
-- 
1.8.0

